Fix type guard errors in batch and block commands

@claude Please add proper type guards to handle Partial vs Full response types

**Problem:**
The Notion API can return either Full or Partial response types. Code is trying to access properties that only exist on Full types without checking first.

**Compilation errors:**

**1. batch/retrieve.ts lines 124, 127 - Partial types assigned to Full types**
```
src/commands/batch/retrieve.ts(124,11): error TS2322: Type 'GetPageResponse' is not assignable to type 'PageObjectResponse | BlockObjectResponse | GetDataSourceResponse'.
  Type 'PartialPageObjectResponse' is not assignable...
  
src/commands/batch/retrieve.ts(127,11): error TS2322: Type 'GetBlockResponse' is not assignable to type 'PageObjectResponse | BlockObjectResponse | GetDataSourceResponse'.
  Type 'PartialBlockObjectResponse' is not assignable...
```

**2. batch/retrieve.ts line 247 - Type comparison issue**
```
src/commands/batch/retrieve.ts(247,24): error TS2367: This comparison appears to be unintentional because the types '"data_source" | "block"' and '"database"' have no overlap.
```

**3. block/update.ts lines 191-197 - Accessing .type without type guard**
```
src/commands/block/update.ts(191,49): error TS2339: Property 'type' does not exist on type 'GetBlockResponse'.
src/commands/block/update.ts(192,79): error TS2339: Property 'type' does not exist on type 'GetBlockResponse'.
src/commands/block/update.ts(196,22): error TS2339: Property 'type' does not exist on type 'GetBlockResponse'.
src/commands/block/update.ts(197,27): error TS2339: Property 'type' does not exist on type 'GetBlockResponse'.
```

**Solution:**
Use `isFullPage()` and `isFullBlock()` type guards from `@notionhq/client` before accessing properties.

**Example:**
```typescript
import { isFullPage, isFullBlock } from '@notionhq/client'

const response = await retrievePage({ page_id: pageId })
if (\!isFullPage(response)) {
  throw new Error('Got partial page response')
}
// Now TypeScript knows response has all properties
const title = response.properties.title
```

**Tasks:**
1. In `src/commands/batch/retrieve.ts`:
   - Add type guards for page responses (line 124)
   - Add type guards for block responses (line 127)
   - Fix the type comparison at line 247 (check what types are being compared)
2. In `src/commands/block/update.ts`:
   - Add `isFullBlock()` type guard before accessing `.type` property at lines 191-197

**Files to modify:**
- `src/commands/batch/retrieve.ts` (add type guards for lines 124, 127, fix line 247)
- `src/commands/block/update.ts` (add type guard before line 191)

**Reference:**
- Type guards from `@notionhq/client`: `isFullPage()`, `isFullBlock()`, `isFullDatabase()`
- Example usage in other commands that already use type guards correctly

--- a/src/commands/batch/retrieve.ts
+++ b/src/commands/batch/retrieve.ts
@@ -14,6 +14,7 @@ import {
   wrapNotionError
 } from '../../errors'
 import { PageObjectResponse, BlockObjectResponse, GetDataSourceResponse } from '@notionhq/client/build/src/api-endpoints'
+import { isFullPage, isFullBlock } from '@notionhq/client'
 import * as readline from 'readline'
 
 type RetrieveResult = {
@@ -120,12 +121,32 @@ export default class BatchRetrieve extends Command {
       let data: PageObjectResponse | BlockObjectResponse | GetDataSourceResponse
 
       switch (type) {
-        case 'page':
-          data = await notion.retrievePage({ page_id: id })
+        case 'page': {
+          const pageResponse = await notion.retrievePage({ page_id: id })
+          if (!isFullPage(pageResponse)) {
+            throw new NotionCLIError(
+              NotionCLIErrorCode.API_ERROR,
+              'Received partial page response instead of full page',
+              [],
+              { attemptedId: id }
+            )
+          }
+          data = pageResponse
           break
-        case 'block':
-          data = await notion.retrieveBlock(id)
+        }
+        case 'block': {
+          const blockResponse = await notion.retrieveBlock(id)
+          if (!isFullBlock(blockResponse)) {
+            throw new NotionCLIError(
+              NotionCLIErrorCode.API_ERROR,
+              'Received partial block response instead of full block',
+              [],
+              { attemptedId: id }
+            )
+          }
+          data = blockResponse
           break
+        }
         case 'database':
           data = await notion.retrieveDataSource(id)
           break
@@ -244,7 +265,7 @@ export default class BatchRetrieve extends Command {
           if ('object' in result.data) {
             if (result.data.object === 'page') {
               title = getPageTitle(result.data as PageObjectResponse)
-            } else if (result.data.object === 'database') {
+            } else if (result.data.object === 'data_source') {
               title = getDataSourceTitle(result.data as GetDataSourceResponse)
             } else if (result.data.object === 'block') {
               title = getBlockPlainText(result.data as BlockObjectResponse)

--- a/src/commands/block/update.ts
+++ b/src/commands/block/update.ts
@@ -4,6 +4,7 @@ import {
   UpdateBlockParameters,
   BlockObjectResponse,
 } from '@notionhq/client/build/src/api-endpoints'
+import { isFullBlock } from '@notionhq/client'
 import { outputRawJson, getBlockPlainText, buildBlockUpdateFromTextFlags } from '../../helper'
 import { resolveNotionId } from '../../utils/notion-resolver'
 import { AutomationFlags } from '../../base-flags'
@@ -179,7 +180,17 @@ export default class BlockUpdate extends Command {
       // Handle color updates
       if (flags.color) {
         // Retrieve the block to determine its type
-        const block = await notion.retrieveBlock(blockId)
+        const blockResponse = await notion.retrieveBlock(blockId)
+
+        // Ensure we have a full block response
+        if (!isFullBlock(blockResponse)) {
+          throw new NotionCLIError(
+            NotionCLIErrorFactory.apiError('block', blockId).code,
+            'Received partial block response. Cannot determine block type for color update.',
+            [],
+            { attemptedId: blockId }
+          )
+        }
 
         // Color is only supported for certain block types
         const colorSupportedTypes = [
@@ -188,13 +199,13 @@ export default class BlockUpdate extends Command {
           'quote', 'callout'
         ]
 
-        if (!colorSupportedTypes.includes(block.type)) {
-          this.error(`Color property is not supported for block type: ${block.type}. Supported types: ${colorSupportedTypes.join(', ')}`)
+        if (!colorSupportedTypes.includes(blockResponse.type)) {
+          this.error(`Color property is not supported for block type: ${blockResponse.type}. Supported types: ${colorSupportedTypes.join(', ')}`)
         }
 
         // Color must be nested within the block type property
-        params[block.type] = {
-          ...params[block.type],
+        params[blockResponse.type] = {
+          ...params[blockResponse.type],
           color: flags.color
         }
       }

